server_addon/blender/client/ayon_blender/plugins/publish/extract_abc.py

```diff
@@ -44,6 +44,20 @@ def process(self, instance):
         context = plugin.create_blender_context(
             active=asset_group, selected=selected)
 
+        scene = bpy.context.scene
+        frame_start = scene.frame_start
+        frame_end = scene.frame_end
+        frame_step = scene.frame_step
+        fps = scene.render.fps
+        fps_base = scene.render.fps_base
+        scene.frame_start = instance.data.get("frameStart", frame_start)
+        scene.frame_end = instance.data.get("frameEnd", frame_end)
+        scene.frame_step = instance.data.get("frameStep", frame_step)
+        inst_fps = instance.data.get("fps")
+        if inst_fps:
+            scene.render.fps = inst_fps
+            scene.render.fps_base = 1
+
         with bpy.context.temp_override(**context):
             # We export the abc
             bpy.ops.wm.alembic_export(
```

Comment:
The [alembic export command](https://docs.blender.org/api/current/bpy.ops.wm.html#bpy.ops.wm.alembic_export) can take a `start` and `end`? Why do we need to change the scene?

Additionally, I would avoid exposing the FPS on the instance for this if it's this tied to the scene. Coming from maya where often changing the FPS has so much influence on the scene that it hardly makes a reproducable output file. But maybe that's different in Blender?

If we do need to change the scene to export the way we want - then we should use a context manager!

Also, if the logic is this involved and we're duplicating it this many times we should make it into a library function.